src/ui/pages/SwapForm/shared/price-impact.ts

```diff
@@ -1,11 +1,84 @@
-import type BigNumber from 'bignumber.js';
+import type { Asset } from 'defi-sdk';
+import BigNumber from 'bignumber.js';
+import { isNumeric } from 'src/shared/isNumeric';
+import { formatPercent } from 'src/shared/units/formatPercent/formatPercent';
 
-const PRICE_IMPACT_WARNING_THRESHOLD = -0.05;
+const LOW_LOSS_RATIO = -0.05;
+const HIGH_LOSS_RATIO = -0.15;
 
-export function exceedsPriceImpactThreshold({
-  relativeChange,
+type PriceImpactLevel = 'low' | 'medium' | 'high';
+
+export type PriceImpact =
+  | { kind: 'n/a' }
+  | {
+      kind: 'profit';
+      ratio: BigNumber;
```

Comment:
If I understand it correctly, we don't need `ratio` field in the result object.

Reply:
Also what do you think about just make object with 2 props:
```
{
  kind: 'loss' | 'big-loss' | 'neutral'
  percentage: number
}
```
?

It should cover all cases from the task with less cases in types.

Reply:
I chose to retain the `ratio` number as it provides context for understanding the derivation of percentage and other values. It serves as the source data for the percentage (which is primarily for visual purposes rather than conditional logic). Since it's simply a numeric value (carrying it around should't impact performance that much) and might be useful for potential future conditional logic, I decided to keep it

Reply:
just read [this comment](https://github.com/zeriontech/zerion-wallet-extension/pull/740#discussion_r2007847769) and now I'm thinking about removing `percentage` formatted value, but keeping the `ratio` :)